adam-core/src/main/scala/org/bdgenomics/adam/rdd/ShuffleRegionJoin.scala

```diff
@@ -369,38 +267,41 @@ private trait VictimlessSortedIntervalPartitionJoin[T, U, RU]
     with Serializable {
 
   // stores the rightIter values that might overlap the current value from the leftIter
-  private var cache: ListBuffer[(ReferenceRegion, U)] = ListBuffer.empty
+  private val cache: ListBuffer[(ReferenceRegion, U)] = ListBuffer.empty
 
   protected def postProcessHits(iter: Iterator[(T, U)],
                                 currentLeft: T): Iterator[(T, RU)]
 
-  protected def advanceCache(until: Long): Unit = {
-    while (right.hasNext && right.head._1._1.start < until) {
+  protected def advanceCache(until: ReferenceRegion): Unit = {
+    while (right.hasNext && (right.head._1.compareTo(until) <= 0 || right.head._1.covers(until))) {
       val x = right.next()
-      cache += x._1._1 -> x._2
+      cache += x._1 -> x._2
     }
   }
 
-  protected def pruneCache(to: Long) {
-    cache = cache.dropWhile(_._1.end <= to)
+  protected def pruneCache(to: ReferenceRegion) {
+    cache --= cache.takeWhile(f => f._1.compareTo(to) < 0 && !f._1.covers(to))
```

Comment:
OOC, why's this rewritten as `takeWhile` then remove, instead of just `dropWhile`?